.gitlab/container_build/docker_linux.yml

```diff
@@ -89,6 +89,33 @@ docker_build_agent7_arm64:
     TAG_SUFFIX: -7
     BUILD_ARG: --target test --build-arg PYTHON_VERSION=3 --build-arg DD_AGENT_ARTIFACT=datadog-agent-7*-arm64.tar.xz
 
+# build agent7 fips image
+docker_build_fips_agent7:
+  extends: [.docker_build_job_definition_amd64, .docker_build_artifact]
+  rules:
+    - !reference [.except_mergequeue]
+    - when: on_success
+  needs:
+    - job: datadog-agent-7-x64-fips
+  variables:
+    IMAGE: registry.ddbuild.io/ci/datadog-agent/agent
+    BUILD_CONTEXT: Dockerfiles/agent
+    TAG_SUFFIX: -7-fips
+    BUILD_ARG: --target test --build-arg PYTHON_VERSION=3 --build-arg DD_AGENT_ARTIFACT=datadog-fips-agent-7*-amd64.tar.xz
```

Comment:
Just a note, this is likely to conflict with https://github.com/DataDog/datadog-agent/pull/31483
Nothing critical but better be aware

Reply:
fixed in https://github.com/DataDog/datadog-agent/pull/31004/commits/4095a2e65be792f7ee9fd3699a16521d2c75b397